### PR TITLE
FEAT-#5880: add `to_sql` parallel implementation for Dask

### LINKED ISSUE #5880: FEAT: add `to_sql` parallel implementation for Dask

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

Let's think about code duplication after a new release.

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5880 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Review by mvashishtha — COMMENTED** [review]
@anmyachev do we need to update "supported APIs"?

**Comment by anmyachev** [comment]
> @anmyachev do we need to update "supported APIs"?

No, we don't have any clarification on which engine supports `to_sql`. Just "Y".

**Comment by mvashishtha** [comment]
@anmyachev there's  a merge conflict with master.

**Review by mvashishtha — APPROVED** [review]

**Comment by anmyachev** [comment]
> @anmyachev there's a merge conflict with master.

@mvashishtha thanks! Fixed

**Review by mvashishtha — APPROVED** [review]

**Review by mvashishtha — APPROVED** [review]

**Review by vnlitvinov — APPROVED** [review]
LGTM!

File: modin/core/execution/ray/implementations/pandas_on_ray/io/io.py [hunk 1]
```diff
@@ -112,8 +112,7 @@ def func(df):
         # Ensure that the metadata is syncrhonized
```
Comment (anmyachev) [inline modin/core/execution/ray/implementations/pandas_on_ray/io/io.py:112]:
```suggestion
        # Ensure that the metadata is synchronized
```

File: modin/core/execution/unidist/implementations/pandas_on_unidist/io/io.py [hunk 2]
```diff
@@ -117,8 +117,7 @@ def func(df):
         # Ensure that the metadata is syncrhonized
```
Comment (anmyachev) [inline modin/core/execution/unidist/implementations/pandas_on_unidist/io/io.py:117]:
```suggestion
        # Ensure that the metadata is synchronized
```

**Comment by anmyachev** [comment]
@mvashishtha ready to merge

**Review by mvashishtha — APPROVED** [review]

**Pull request merged**